stl/src/StlCompareStringA.cpp

```diff
@@ -80,8 +80,7 @@ extern "C" int __cdecl __crtCompareStringA(_In_z_ LPCWSTR LocaleName, _In_ DWORD
             }
 
             for (auto cp = &cpInfo.LeadByte[0]; cp[0] && cp[1]; cp += 2) {
-                const auto ch = static_cast<unsigned char>(*lpString1);
-                if (ch >= cp[0] && ch <= cp[1]) {
+                if (const auto ch = static_cast<unsigned char>(*lpString1); ch >= cp[0] && ch <= cp[1]) {

```

Comment:
Please don't do this. The reason it works is because of the semicolon. The c++ standard states that you can't do declaration and boolean logic in an if statement. I don't know why you added this to the if statement.

Reply:
Actually, you can.
```cpp
if (const auto foo = something(); foo.bar())
{
}
```
is perfectly valid C++. However in this case it does not improve the code, it makes it harder to read IMO.

Reply:
Actually it is available in C++20 but we have to stay compatible with C++14.

Reply:
> The c++ standard states that you can't do declaration and boolean logic in an if statement.

This is allowed since C++17. BTW, you can write `if (int x = y) {}` even in ancient C++98.